### PR TITLE
Replacing DB ADAL token cache with an in memory token cache

### PR DESCRIPTION
**Pull request by shklar**
1. Replacing DB ADAL token cache with an in memory token cache.
2. Removing Shared app config (wasn't used)

### CODE REVIEW

File: Shared/ADALTokenCache.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ namespace CogsMinimizer.Shared
 {
     public class ADALTokenCache : TokenCache
     {
-        private DataAccess db = new DataAccess();
+        private InMemoryTokenCache memoryCache = InMemoryTokenCache.Instance;
```
Comment (bergano65) [inline Shared/ADALTokenCache.cs]: if we stopped use database storing tokens why we  needed additional  class in memory token cache and do not implement all needed functionality here in one class?

File: Shared/ADALTokenCache.cs [hunk 2]
```diff
@@ -84,24 +83,25 @@ void AfterAccessNotification(TokenCacheNotificationArgs args)
...
+  
                 if (Cache == null)
                 {
                     // if no existing entry for that user, create a new one
                     Cache = new PerUserTokenCache
                     {
                         webUserUniqueId = User,
                     };
+
+                    //Add to memory cache for later use
```
Comment (bergano65) [inline Shared/ADALTokenCache.cs]: "// Add" please add space for readability

File: Shared/ADALTokenCache.cs [hunk 3]
```diff
@@ -84,24 +83,25 @@ void AfterAccessNotification(TokenCacheNotificationArgs args)
...
 
                 // update the cache contents and the last write timestamp
                 Cache.cacheBits = this.Serialize();
-                Cache.LastWrite = DateTime.Now;
+                Cache.LastWrite = DateTime.UtcNow;
 
-                // update the DB with modification or new entry
-                db.Entry(Cache).State = Cache.Id == 0 ? EntityState.Added : EntityState.Modified;
-                db.SaveChanges();
+                //Cache change has been recorded. Resetting the flag.
```
Comment (bergano65) [inline Shared/ADALTokenCache.cs]: add space please for readability

File: Shared/ADALTokenCache.cs (hunk 1 above)
Comment (shklar) [reply Shared/ADALTokenCache.cs]: The ADALTokenCache is a client side library for a single user.
I have replaced the server side implementation from a DB to in memory to serve multiple users.

File: Shared/ADALTokenCache.cs (hunk 3 above)
Comment (shklar) [reply Shared/ADALTokenCache.cs]: Done

File: Shared/ADALTokenCache.cs (hunk 2 above)
Comment (shklar) [reply Shared/ADALTokenCache.cs]: Done

File: Shared/ADALTokenCache.cs [hunk 4]
```diff
@@ -35,7 +35,7 @@ public ADALTokenCache(string user)
             this.BeforeWrite = BeforeWriteNotification;
 
             // look up the entry in the DB
-            Cache = db.PerUserTokenCacheList.FirstOrDefault(c => c.webUserUniqueId == User);
+            Cache = memoryCache.PerUserTokenCacheList.FirstOrDefault(c => c.webUserUniqueId == User);
```
Comment (ischrei) [inline Shared/ADALTokenCache.cs]: Memory cache means it wont be optimal if we have multiple instances, but I guess it's good enough for now.

File: Shared/App.config [hunk 5]
```diff
@@ -1,31 +0,0 @@
...
-    <add key="UnobtrusiveJavaScriptEnabled" value="true" />
-    <add key="ida:Authority" value="https://login.windows.net/{0}/" />
-    <add key="ida:Issuer" value="https://sts.windows.net/{0}/" />
-    <add key="ida:GraphAPIIdentifier" value="https://graph.windows.net/" />
-    <add key="ida:GraphAPIVersion" value="1.5" />
-    <add key="ida:AzureResourceManagerIdentifier" value="https://management.core.windows.net/" />
-    <add key="ida:AzureResourceManagerUrl" value="https://management.azure.com" />
-    <add key="ida:AzureResourceManagerAPIVersion" value="2014-04-01-preview" />
-    <add key="ida:ARMAuthorizationPermissionsAPIVersion" value="2014-07-01-preview" />
-    <add key="ida:ARMAuthorizationRoleDefinitionsAPIVersion" value="2014-07-01-preview" />
```
Comment (ischrei) [inline Shared/App.config]: don't we need those settings?

Comment (shklar) [reply Shared/App.config]: Evgeny is handling this in his PR.

File: Shared/ADALTokenCache.cs [hunk 6]
```diff
@@ -71,7 +70,7 @@ select new
                 if (status.Count() > 0 && status.First().LastWrite > Cache.LastWrite)
                 //// read from from storage, update in-memory copy
                 {
-                    Cache = db.PerUserTokenCacheList.FirstOrDefault(c => c.webUserUniqueId == User);
```
[the remark this answers is not in the document]
Comment (shklar) [reply Shared/ADALTokenCache.cs]: I will be deleting the table from the DBs when this change is deployed.

**Pull request merged**